app/src/main/java/cm/aptoide/pt/network/model/IdsRepository.kt

```diff
@@ -0,0 +1,99 @@
+package cm.aptoide.pt.network.model
+
+import android.content.Context
+import android.os.Looper
+import androidx.annotation.WorkerThread
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.google.android.gms.ads.identifier.AdvertisingIdClient
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.runBlocking
+import rx.schedulers.Schedulers
+import timber.log.Timber
+import java.util.UUID
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class IdsRepository @Inject constructor(
+  private val context: Context,
+  private val dataStore: DataStore<Preferences>,
+) {
+
+  @Synchronized fun getUniqueIdentifier(): rx.Single<String> {
+    val aptoideId: String = getAndroidClientUUID()
```

Comment:
Why do you need rxjava here ?

Reply:
i've removed RxJava, now working with normal flows and datastores